plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/expression/ImplementAvg.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.query.expression;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.pinot.PinotColumnHandle;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.Type;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.variable;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static java.lang.String.format;
+
+public class ImplementAvg
+        implements AggregateFunctionRule
+{
+    // Extracted from io.trino.plugin.jdbc.expression
```

Comment:
Remove the comment since it's normal for each datasource to have their own aggregation function semantics and hence the rewrite rules need to be different too.

Reply:
Same for other rewrites too IMO. (But yeah, I would like to use the changes in https://github.com/trinodb/trino/pull/8578 if possible).